legacy/reqcounter/reqcounter.go

```diff
@@ -19,76 +19,124 @@ package reqcounter
 import (
 	"fmt"
 	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/sirupsen/logrus"
 )
 
 // RequestCounter records the number of HTTP requests to GCR.
+// TODO: @tylerferrara in the future, add a field 'persistent bool' to determine if
+// the counter should ever reset.
 type RequestCounter struct {
-	mutex    sync.Mutex // Lock to prevent race-conditions with concurrent processes.
-	requests uint64     // Number of HTTP requests since recording started.
-	since    time.Time  // When the current counter began recording requests.
+	Mutex    sync.Mutex    // Lock to prevent race-conditions with concurrent processes.
+	Requests uint64        // Number of HTTP requests since recording started.
+	Since    time.Time     // When the current counter began recording requests.
+	Interval time.Duration // The duration of time between each log.
 }
 
-// log records the number of HTTP requests found and resets the counter.
-func (rc *RequestCounter) log() {
+// increment adds 1 to the request counter, signifying another call to GCR.
+func (rc *RequestCounter) Increment() {
+	rc.Mutex.Lock()
+	atomic.AddUint64(&rc.Requests, 1)
+	rc.Mutex.Unlock()
+}
+
+// Flush records the number of HTTP requests found and resets the counter.
+func (rc *RequestCounter) Flush() {
 	// Hold onto the lock when reading & writing the counter.
-	rc.mutex.Lock()
-	defer rc.mutex.Unlock()
+	rc.Mutex.Lock()
+	defer rc.Mutex.Unlock()
 
 	// Log the number of requests within this measurement window.
-	msg := fmt.Sprintf("Since %s there have been %d requests to GCR.", counter.since.Format("2006-01-02 15:04:05"), rc.requests)
-	logrus.Debug(msg)
+	msg := fmt.Sprintf("As of %s, within %s min, there have been %d requests to GCR.", rc.Interval, rc.Since.Format("2006-01-02 15:04:05"), rc.Requests)
```

Comment:
I think you have rc.Interval and rc.Since reversed. But also, can you make this more explicit? Something like this:

"From _start_timestamp to _end_timestamp [_interval], there have been %d requests to GCR."

You'd have to do the _end_timestamp calculation but that's simple because it's just `time.Now()`. I think it's easier to read.

Reply:
Absolutely! That reads much nicer.

Reply:
For additional readability, intervals will be displayed in "minutes".